osbenchmark/builder/launchers/docker_launcher.py

```diff
@@ -0,0 +1,77 @@
+import logging
+import os
+
+from osbenchmark import time, telemetry
+from osbenchmark.builder import cluster
+from osbenchmark.builder.launchers.launcher import Launcher
+from osbenchmark.exceptions import LaunchError, ExecutorError
+
+
+class DockerLauncher(Launcher):
+    # May download a Docker image and that can take some time
+    PROCESS_WAIT_TIMEOUT_SECONDS = 10 * 60
+
+    def __init__(self, pci, shell_executor, clock=time.Clock):
+        super().__init__(shell_executor)
+        self.logger = logging.getLogger(__name__)
+        self.clock = clock
+
+    def start(self, host, node_configurations):
+        nodes = []
+        for node_configuration in node_configurations:
+            node_name = node_configuration.node_name
+            host_name = node_configuration.ip
+            binary_path = node_configuration.binary_path
+            self.logger.info("Starting node [%s] in Docker.", node_name)
+            self._start_process(host, binary_path)
+            node_telemetry = [
+                # Don't attach any telemetry devices for now but keep the infrastructure in place
+            ]
+            t = telemetry.Telemetry(devices=node_telemetry)
+            node = cluster.Node(0, binary_path, host_name, node_name, t)
+            t.attach_to_node(node)
+            nodes.append(node)
+        return nodes
+
+    def _start_process(self, host, binary_path):
+        compose_cmd = self._docker_compose(binary_path, "up -d")
+
+        try:
+            self.shell_executor.execute(host, compose_cmd)
+        except ExecutorError as e:
+            raise LaunchError("Exception starting OpenSearch Docker container", e)
+
+        container_id = self._get_container_id(host, binary_path)
+        self._wait_for_healthy_running_container(host, container_id, DockerLauncher.PROCESS_WAIT_TIMEOUT_SECONDS)
+
+    def _docker_compose(self, compose_config, cmd):
+        return "docker-compose -f {} {}".format(os.path.join(compose_config, "docker-compose.yml"), cmd)
```

Comment:
Nit: ```.format()``` works fine but I think it becomes quite verbose when working with multiple parameters such as this line. Even if this is pre-existing code, it might be worth switching areas to use f-strings and f-strings from here on out for newer snippets of code. f-strings would enhance readability, flexibility, and _slight_ speed over %s and .format(). Again, not necessary, but I think it's worth using f-strings from here on out but I understand if we want to maintain consistency by using .format()

Reply:
I didn't put much time into refactoring inherited code unless it was necessary. I am in agreement that much of it could be cleaned up so I'm happy to refactor stuff like this as folks point it out. 

Will refactor this instance